### PR TITLE
Better multithreading logging

### PR DESCRIPTION
**Pull request by Bod9001**
this should now properly is Display in the discord web hook,

note I removed the throw from the thread because, the other try catches didn't have it

### CODE REVIEW

File: UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs [hunk 1]
```diff
@@ -82,7 +82,15 @@ private void ThreadLoop()
...
 			while (running && threadMode == ThreadMode.Threaded && midTick == false)
 			{
-				RunTick();
+				try
+				{
+					RunTick();
+				}
+				catch (Exception e)
+				{
+					ThreadLogger.AddLog(e.ToString(), Category.Threading);
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs:91]: throw as well so we notice stuff breaking

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs:91]: atmospherics and electrical, both didn't have throw and those were the most likely to generate a lot of errors so

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs:91]: I'm just saying it is fine to catch to log, but you should throw the exception at the end of the handling

Comment (Bod9001) [reply UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs:91]: But the system that would have varied a lot already just had a try catch, with no throw, and the server has been fine, do you think it requires the throw being added?

Comment (corp-0) [reply UnityProject/Assets/Scripts/Core/Threading/ThreadBehaviour.cs:91]: I would rather have it so we can see stuff breaking if it breaks. You first see the efect of stuff breaking then you read the logs.

**Review by corp-0 — APPROVED** [review]

**Pull request merged**